src/GraphQL.SystemReactive/SubscriptionExecutionStrategy.cs

```diff
@@ -128,12 +128,19 @@ await listener.AfterExecutionAsync(context)
                             }
                         }
 
+                        // Set the execution node's value to null if necessary
+                        executionNode.ClearErrorNodes();
```

Comment:
If it returns `true` ? In `ExecutionStrategy` you use `  object data = rootNode.ClearErrorNodes() ? null : rootNode;`

Reply:
I thought long and hard about this.  Pretty sure it's correct.  Let's say there's a problem.  (Ignoring the fact that errors are not being returned, not sure why.)  What happens is that the executionNode will clear itself so ToValue will return null.  The Root node will then serialize this result:

```json
{
  "cart": null
}
```

rather than what it would normally do:

```json
{
  "cart": {
    "count": 3
  }
}
```

So basically the code is designed to assume that the subscription field is nullable.  Maybe we should check the spec on that.

Reply:
I can't find it in the spec (although it's probably there somewhere).  I've never used subscriptions so if you have any insight, let me know.  I'd feel pretty confident that the implementation is correct, and also, it matches what was there before.

Reply:
Let's keep `//TODO: check` with a link to this comment https://github.com/graphql-dotnet/graphql-dotnet/pull/2240#discussion_r570631402 in the code. We'll check it someday.

Reply:
So I guess a subscription field can be defined as a non-null object type within GraphQL.Net.  I don't know if that meets spec or not.  Spec says that a subscription field must be an object type.  But let's say that was allowed.  Then the response would be:
```json
{
}
```
Does that make any sense?  How would the receiving end of the subscription know which subscription failed?  I don't know enough about subscriptions, but my gut feeling is that it doesn't make sense.

Reply:
```suggestion
                        // Note: assumes that the subscription field is nullable, regardless of how it was defined
                        // See https://github.com/graphql-dotnet/graphql-dotnet/pull/2240#discussion_r570631402
                        //TODO: check if a non-null subscription field is allowed per the spec
                        //TODO: check if errors should be returned along with the data
                        executionNode.ClearErrorNodes();
```

Reply:
> How would the receiving end of the subscription know which subscription failed?

Every subscription has a unique identifier and state bot server and client side.